src/composables/transactionList.ts

```diff
@@ -129,149 +127,75 @@ export function useTransactionList({ store }: IDefaultComposableOptions) {
-      );
-      if (response.message) {
-        return [];
-      }
-      // TODO prepare interface for response
-      const tipWithdrawnTransactions: ITransaction[] = (response as any[]).map(({
-        amount,
-        contract,
-        height,
-        data: { tx },
-        ...t
-      }) => ({
-        tx: {
-          ...tx,
-          address,
-          amount,
-          contractId: contract,
-          type: Tag[Tag.ContractCallTx],
-        },
-        ...t,
-        microTime: new Date(t.createdAt).getTime(),
-        blockHeight: height,
-        claim: true,
-      }));
-
-      return tipWithdrawnTransactions;
-    } catch (error) {
-      return [];
-    }
-  }
-
   async function fetchTransactions(
-    limit: number,
-    recent: boolean,
     address: Encoded.AccountAddress,
+    protocol?: Protocol,
   ): Promise<ITransaction[]> {
-    await getAeSdk(); // Ensure the `nodeNetworkId` is established
-
+    await getAeSdk();
     ensureAccountTransactionStateExists(address);
     const transactionState = transactions.value[address];
-
-    if (transactionState.nextPageUrl === null && !recent) {
+    if (!protocol) {
       return [];
     }
+    const recent = transactionState.nextPageUrl === '';
+
+    const adapter = ProtocolAdapterFactory.getAdapter(protocol);
 
-    const [
-      regularTransactions,
-      pendingTransactions,
-      tipWithdrawnTransactions,
-    ] = await Promise.all([
-      fetchTransactionsFromMiddleware(transactionState, address, recent, limit),
-      fetchPendingTransactions(address),
-      fetchTipWithdrawnTransactions(address, recent),
-    ]);
+    const response = await adapter.fetchTransactions(
+      address,
+      transactionState.nextPageUrl,
+    );
+
+    const pendingTransactions = response.pendingTransactions?.filter(
+      (transaction: ITransaction) => !transactionState.pending[nodeNetworkId.value!]?.find(
+        (tx) => tx?.hash === transaction?.hash,
+      ),
+    ).map((transaction: ITransaction) => ({ ...transaction, pending: true })) || [];
 
     const lastPendingTransaction = pendingTransactions?.[pendingTransactions.length - 1];
+
     if (lastPendingTransaction?.type === AEX9_TRANSFER_EVENT) {
       const middleware = await getMiddleware();
       pendingTransactions[pendingTransactions.length - 1] = (
-        await middleware.getTx(lastPendingTransaction.payload.txHash)
+        await middleware.getTx((lastPendingTransaction as any)?.payload.txHash)
       );
     }
 
+    setTransactionsNextPage(address as string, response.nextPageParams);
+
     let preparedTransactions = [
       ...pendingTransactions,
-      ...regularTransactions,
+      ...response.regularTransactions,
     ]
       .filter(({ type }) => !type?.startsWith('Internal'))
       .map((transaction) => ({
-        ...(transaction.payload || transaction),
+        ...((transaction as any).payload || transaction),
```

Comment:
don't we have a type for transaction? If you'd type it in the map you won't need typing below